course02/homework02/danielellis/SlidingWindow.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include <map>
+#include <queue>
+#include <chrono>
+#include <vector>
+
+#include "Buffer.h"
+
+template<class Message>
+class SlidingWindow
+{
+public:
+	using value_type = Message;
+
+	SlidingWindow(const WindowConfig& conf);
+	void Send(int clientId, const Message& message);
+	std::vector<Message> DumpMessages(int clientId) const;
```

Comment:
This is potentially a very expensive operation as you would copy all messages. A better patern is to visit elements (visitor).